src/GitHubDeploymentManager.ts

```diff
@@ -300,6 +309,80 @@ export class GitHubDeploymentManager {
     });
   }
 
+  async fetchLabelsForIssue(issueId: number): Promise<string[]> {
+    core.info(`Fetching labels for issue ${issueId}`);
+    return this.github.rest.issues.listLabelsOnIssue({
+      ...this.repo,
+      issue_number: issueId,
+      per_page: 100,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    }).then(resp => {
+      return resp.data.map(label => label.name);
+    }).catch(() => {
+      return [];
+    })
+  }
+
+  async fetchLabelsForRepo(
+    state: keyof typeof ISSUE_STATES = ISSUE_STATES.open,
```

Comment:
As described in my comment for `constants.ts`, let's not use a pseudo-enum here but let's keep things simple:

Derive the issue-state type from the octokit types and then just use a string for the default definition here. it would look something like this:

```typescript
  type OctokitType = InstanceType<typeof Octokit>;

	type Issue = Awaited<
		ReturnType<OctokitType["rest"]["issues"]["listForRepo"]>
	>["data"][0];
	type IssueState = Issue["state"];

  [...]
  async fetchLabelsForRepo(
    state: IssueState = 'open'
  [...]
```